From 9589cffddf6d6f99522cd8f46a360a73bad1e068 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Wed, 19 Oct 2016 20:28:26 +0100
Subject: [PATCH 3/6] wm/ibus: Always try to use an existing ibus daemon if
 possible

As per issue #661, we should ensure we never miss a connected signal,
and make sure we check for the existence of a running daemon prior to
doing anything.

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 wm/ibus.vala | 11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/wm/ibus.vala b/wm/ibus.vala
index 1676c17..242b06d 100644
--- a/wm/ibus.vala
+++ b/wm/ibus.vala
@@ -57,15 +57,20 @@ public class IBusManager : GLib.Object
         this.engines = new HashTable<string,weak IBus.EngineDesc>(str_hash, str_equal);
 
         /* Get the bus */
-        bus = new IBus.Bus.async();
+        bus = new IBus.Bus();
 
         /* Hook up basic signals */
         bus.connected.connect(this.ibus_connected);
         bus.disconnected.connect(this.ibus_disconnected);
         bus.set_watch_dbus_signal(true);
 
-        /* Start the ibus daemon */
-        this.startup_ibus();
+        /* Should have ibus running already */
+        if (bus.is_connected()) {
+            this.ibus_connected();
+        } else {
+            /* Start the ibus daemon */
+            this.startup_ibus();
+        }
     }
 
     /**
-- 
2.10.1

